### PR TITLE
Revert "Remove splitTasks as a temporary fix for unstable index read"

### PR DESCRIPTION
**Pull request by marsishandsome**
Reverts pingcap/tispark#875

cause we find the bug: https://github.com/pingcap/tispark/pull/878

### CODE REVIEW

File: tikv-client/src/main/java/com/pingcap/tikv/operation/iterator/CoprocessIterator.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
 public abstract class CoprocessIterator<T> implements Iterator<T> {
   protected final TiSession session;
   protected final List<RegionTask> regionTasks;
-  protected final DAGRequest dagRequest;
+  protected DAGRequest dagRequest;
```
Comment (birdstorm) [inline tikv-client/src/main/java/com/pingcap/tikv/operation/iterator/CoprocessIterator.java]: maybe we can leave this change.

Comment (marsishandsome) [reply tikv-client/src/main/java/com/pingcap/tikv/operation/iterator/CoprocessIterator.java]: maybe in other PR to add `final`

**Pull request merged**